test/integration/IntegrationTest.cpp

```diff
@@ -256,26 +266,51 @@ void IntegrationTest::analyzeRecording(const std::vector<int16_t>& recording,
         return;
     }
 
-    for (size_t cursor = 0; cursor < size - fftWindowSize; cursor += 256)
-    {
-        for (uint64_t x = 0; x < fftWindowSize; ++x)
-        {
-            testVector[x] = std::complex<double>(static_cast<double>(recording[x + cursor]), 0.0) / (256.0 * 128);
-        }
+    auto start = std::chrono::high_resolution_clock::now();
 
-        SampleDataUtils::fft(testVector);
+    auto const numThreads = getNumWorkingThreads();
+    std::vector<std::thread> workers;
+    std::mutex fft_mutex;
 
-        std::vector<double> frequencies;
-        SampleDataUtils::listFrequencies(testVector, codec::Opus::sampleRate, frequencies);
-        for (size_t i = 0; i < frequencies.size() && i < 50; ++i)
-        {
-            if (std::find(frequencyPeaks.begin(), frequencyPeaks.end(), frequencies[i]) == frequencyPeaks.end())
+    for (size_t threadId = 0; threadId < numThreads; threadId++)
+    {
+        workers.push_back(std::thread([&frequencyPeaks, &recording, &fft_mutex, &logId, size, numThreads, threadId]() {
```

Comment:
I have read multiple times. I still do not like that we create a fat lambda starting at 3 indentation levels.
Break this out into a method and pass that one to the thread.